src/main/kotlin/com/hypto/iam/server/service/UsersService.kt

```diff
@@ -150,6 +152,22 @@ class UsersServiceImpl : KoinComponent, UsersService {
         val userHrn = ResourceHrn(organizationId, "", IamResources.USER, userName)
         return getUser(userHrn, user)
     }
+
+    override suspend fun authenticate(userName: String, password: String): User {
+        if (!appConfig.app.uniqueUsersAcrossOrganizations)
+            throw BadRequestException("Email not unique across organizations. " +
+                "Please use Token APIs with organization ID")
+
+        val userRecord = userRepo.findByEmail(userName)
+            ?: throw UserNotFoundException("Email not found, Unable to authenticate user")
+        val org = organizationRepo.findById(userRecord.organizationId)
+            ?: throw EntityNotFoundException("Invalid organization id name. Unable to authenticate user")
```

Comment:
For EntityNotFoundException, we would be throwing 404. Can we throw a 500 instead if org record is not available because this case can happen only when the data in the table is inconsistent.